library/src/main/java/com/datatorrent/lib/parser/Parser.java

```diff
@@ -47,10 +47,19 @@
  * @since 3.2.0
  */
 @InterfaceStability.Evolving
-public abstract class Parser<INPUT> extends BaseOperator implements Converter<INPUT, Object>,
-    ActivationListener<Context>
+public abstract class Parser<INPUT> extends BaseOperator implements Converter<INPUT, Object>
 {
   protected transient Class<?> clazz;
+  @AutoMetric
+  protected long errorTupleCount;
+  @AutoMetric
+  protected long emittedObjectCount;
+  @AutoMetric
+  protected long rollingPercentValidTuples;
+  @AutoMetric
+  protected long rollingPercentErrorTuples;
+  @AutoMetric
+  protected long incomingTuplesCount;
 
   @OutputPortFieldAnnotation(schemaRequired = true)
```

Comment:
@ishark is correct. If the schemaRequired port is not connected, then validation will not fail even though the TUPLE_CLASS attribute is missing.

@shubham-pathak22 Schema required annotation is an expressive way to let the users know that processing of tuples which are received on this port requires the tuple schema. I will suggest that if you face any limitations or problems with this feature, then you should take it up on apex dev.